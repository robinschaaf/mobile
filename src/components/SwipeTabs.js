import React, { Component } from 'react'
import {
  Image,
  StyleSheet,
  TouchableOpacity,
  View
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import StyledText from './StyledText'
import FieldGuide from './FieldGuide'
import Icon from 'react-native-vector-icons/FontAwesome'
import { length } from 'ramda'

export class ClassifierTabs extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isFieldGuideVisible: false,
    }
  }
  render() {
    const leftButton =
      <TouchableOpacity
        onPress={this.props.onAnswered.bind(this, 1)}
        activeOpacity={0.5}
        style={ [styles.button, styles.tealButton] }>
        <Image source={require('../../images/swipe-left-white.png')} style={styles.swipeIcon} />
        <StyledText additionalStyles={[styles.tealButtonText]} text={ 'No' } />
      </TouchableOpacity>


    const rightButton =
      <TouchableOpacity
        onPress={this.props.onAnswered.bind(this, 0)}
        activeOpacity={0.5}
        style={ [styles.button, styles.tealButton] }>
        <Image source={require('../../images/swipe-right-white.png')} style={styles.swipeIcon} />
        <StyledText additionalStyles={[styles.tealButtonText]} text={ 'Yes' } />
      </TouchableOpacity>

    const fieldGuideButton =
      <TouchableOpacity
        onPress={() => this.setState({isFieldGuideVisible: true})}
        activeOpacity={0.5}
        style={ styles.button }>
        <Icon name='map-o' style={styles.icon} />
        <StyledText additionalStyles={[styles.buttonText]} text={ 'Field Guide' } />
      </TouchableOpacity>

    const fieldGuide =
      <FieldGuide
      guide={this.props.guide}
      isVisible={this.state.isFieldGuideVisible}
      onClose={() => this.setState({isFieldGuideVisible: false})}
      />

    return (
      <View style={styles.container}>
        { leftButton }
        { length(this.props.guide.items) > 0 ? fieldGuideButton : null }
        { rightButton }
        { this.state.isFieldGuideVisible ? fieldGuide : null }
      </View>
    )
  }
}

const styles = EStyleSheet.create({
  $containerHeight: 60,
  container: {
    backgroundColor: '$beckyGray',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    height: '$containerHeight',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  button: {
    backgroundColor: 'white',
    borderColor: '$beckyDisabledIconColor',
    borderWidth: StyleSheet.hairlineWidth,
    borderRadius: 5,
    paddingVertical: 8,
    paddingHorizontal: 12,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  },
  buttonText: {
    color: '$beckyDarkGray',
  },
  tealButton: {
    backgroundColor: '$buttonColor'
  },
  tealButtonText: {
    color: 'white'
  },
  icon: {
    fontSize: 16,
    color: '$beckyDarkGray',
    padding: 3,
    marginRight: 5
  },
  title: {
    fontSize: 11,
    color: 'white',
  },
  swipeIcon: {
    height: 27,
    width: 20,
    marginRight: 10,
  },
})

ClassifierTabs.propTypes = {
  tutorial: React.PropTypes.object,
  device: React.PropTypes.object,
}

export default ClassifierTabs
