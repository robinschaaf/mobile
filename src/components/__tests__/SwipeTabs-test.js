import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SwipeTabs from '../SwipeTabs'

const guide = {
  items: [
    {
      title: 'fake_avatar',
      content: 'Nice project',
      icon: '12345',
    },
    {
      title: 'fake_avatar2',
      content: 'Another Nice project',
      icon: '123456',
    },
  ],
  icons:{
    '12345': { src: 'fake_icon.jpg' },
    '123456': { src: 'fake_icon.jpg' }
  }
}

const answers=[
  { label: 'Yes' },
  { label: 'No' },
]

it('renders correctly and with no button if no field guide', () => {
  const tree = renderer.create(
    <SwipeTabs guide={{ }} onAnswered={jest.fn} answers={answers} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a button if there is a field guide', () => {
  const tree = renderer.create(
    <SwipeTabs guide={guide} onAnswered={jest.fn} answers={answers} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
