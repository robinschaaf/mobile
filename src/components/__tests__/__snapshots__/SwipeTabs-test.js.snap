exports[`test renders a button if there is a field guide 1`] = `
<View>
  <View
    style={undefined}>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={
        Array [
          undefined,
          undefined,
        ]
      }>
      <Image
        source={1}
        style={undefined} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        Yes
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        ï‰¸
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        Field Guide
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={
        Array [
          undefined,
          undefined,
        ]
      }>
      <Image
        source={1}
        style={undefined} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        No
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`test renders correctly and with no button if no field guide 1`] = `
<View>
  <View
    style={undefined}>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={
        Array [
          undefined,
          undefined,
        ]
      }>
      <Image
        source={1}
        style={undefined} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        Yes
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={
        Array [
          undefined,
          undefined,
        ]
      }>
      <Image
        source={1}
        style={undefined} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        No
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
